src/variants.js

```diff
@@ -55,89 +97,122 @@ export async function getVariant(name) {
 	return variants.find(variant => variant.name === name);
 }
 
-export const shortForms = /** @type {const} */ ({
-	'Red': 'r',
-	'Yellow': 'y',
-	'Green': 'g',
-	'Blue': 'b',
-	'Purple': 'p',
-	'Teal': 't',
-	'Black': 'k',
-	'Rainbow': 'm',
-	'White': 'w',
-	'Pink': 'i',
-	'Brown': 'n',
-	'Omni': 'o',
-	'Null': 'u',
-	'Prism': 'i'
-});
+export let shortForms = /** @type {string[]} */ (['r', 'y', 'g', 'b', 'p']);
+
+/**
+ * Edits shortForms to have the correct acryonyms.
+ * @param {Variant} variant
+ */
+export async function getShortForms(variant) {
+	const colors = await colours_promise;
+	const abbreviations = [];
+	for (const suitName of variant.suits) {
+		if (['Black', 'Pink', 'Brown'].includes(suitName)) {
+			abbreviations.push(['k', 'i', 'n'][['Black', 'Pink', 'Brown'].indexOf(suitName)]);
+		} else {
+			const abbreviation = colors.find(color => color.name === suitName)?.abbreviation ?? suitName.charAt(0);
+			if (abbreviations.includes(abbreviation.toLowerCase()))
+				abbreviations.push(suitName.toLowerCase().split('').find(char => !abbreviations.includes(char)));
+			else
+				abbreviations.push(abbreviation.toLowerCase());
+
+		}
+	}
+	shortForms = abbreviations;
+}
 
 /**
  * Returns whether the card would be touched by the clue.
  * @param {Identity} card
- * @param {string[]} suits
+ * @param {Variant} variant
  * @param {Omit<Clue, 'target'>} clue
  */
-export function cardTouched(card, suits, clue) {
+export function cardTouched(card, variant, clue) {
 	const { type, value } = clue;
 	const { suitIndex, rank } = card;
-	const suit = suits[suitIndex];
+	const suit = variant.suits[suitIndex];
 
-	if (suit === 'Null')
+	if (suit === 'Null' || suit === 'Dark Null')
 		return false;
-
-	else if (suit === 'Omni')
+	else if (suit === 'Omni' || suit === 'Dark Omni')
 		return true;
 
-
 	if (type === CLUE.COLOUR) {
-		if (suit === 'White')
+		if (suit.match(variantRegexes.whitish)) {
 			return false;
-
-		else if (suit === 'Rainbow')
+		}
+		else if (suit.match(variantRegexes.rainbowish)) {
 			return true;
+		}
+		else if (suit === 'Prism' || suit === 'Dark Prism') {
+			const colourlessCount = variant.suits.filter(s => s.match(rainbowish) || s.match(whitish) || s.match(/Prism/)).length;
```

Comment:
The nice thing about regexes is that they can be combined and checked in a single pass. In JS you can do this:
```js
const whitish = /White|Gray|Light|Null/;
const rainbowish = /Rainbow|Omni/;
const prismish = /Prism/;
const whitishOrRainbowish = new RegExp(whitish.source + '|' + rainbowish.source + '|' + prismish.source);
```
or maybe with more functional-/helper-styled:
```js
function combineRegex(...regexes) {
   return new RegExp(regexes.map(re => re.source).join('|'));
}
variant.suits.filter(s => s.match(combineRegex(whitish, rainbowish, prismish));
```